cpp/include/raft/stats/contingency_matrix.cuh

```diff
@@ -142,17 +142,25 @@ void contingencyMatrix(const T* groundTruth,
  * @param[in]  ground_truth: device 1-d array for ground truth (num of rows)
  * @param[in]  predicted_label: device 1-d array for prediction (num of columns)
  * @param[out] out_mat: output buffer for contingency matrix
- * @param[in]  min_label: Optional, min value in input ground truth array
- * @param[in]  max_label: Optional, max value in input ground truth array
+ * @param[in]  opt_min_label: Optional, min value in input ground truth array
```

Comment:
Thinking about the docs a bit here, this seems a bit ambiguous now that we aren't explicitly constraining the std::optional  type in the arguments. Can you update this to specify explicitly that a `std::optional` is required here? This applied throughout the PR.